hardtunes_importer.user.js

```diff
@@ -0,0 +1,965 @@
+        title: release_heading.childNodes[0].innerText.replace(/\s/g,' '),
+        format: 'Digital Media',
+        packaging: 'None',
+        country: 'XW',
+        status: 'official',
+        language: 'eng',
+        script: 'Latn',
+        type: 'album',
+        urls: [],
+        labels: [],
+        discs: [],
+    };
+
+    for (form of document.getElementsByClassName('form-group')) {
+        if (!form.childNodes[1])
+            continue;
+        var name = form.childNodes[0].innerText.replace(/\s/g,' ');
+        var value = form.childNodes[1].innerText.replace(/\s/g,' ');
+        switch (name) {
+            case 'Style':
+                style = value;
+                break;
+            case 'Code':
+                if (isNaN(value)) 
+                    catno = value;
+                else
+                    release.barcode = value;
+                break;
+            case 'Release date':
+                release.day = value.split('.')[0];
+                release.month = value.split('.')[1];
+                release.year = value.split('.')[2];
+                break;
+        }
+    }
+
+    // URLs
+    release.urls.push({
+        url: release_url,
+        link_type: MBImport.URL_TYPES.purchase_for_download,
+    });
+
+    // tracks
+
+    for (item of window.document.getElementsByClassName('release-list-item')) {
+        if (!item.getElementsByClassName('release-list-item-number')[0])
+            continue;
+
+        let track = {
+            number: item.getElementsByClassName('release-list-item-number')[0].innerText.replace(/\s/g,' '),
+            title: item.getElementsByClassName('release-list-item-title')[0].innerText.replace(/\s/g,' '),
+            //duration: "0:00",
+            artist_credit: []
+        };
+        let artists = item.getElementsByClassName('release-list-item-artist')[0].childNodes;
+        // artist name and connecting strings alternate
+        for (let i = 0; i < artists.length; i+=2)
+            track.artist_credit.push(translateArtist(artists[i].innerText.replace(/\s/g,' '), (!(artists[i+1])?'':artists[i+1].textContent.replace(/\s/g,' '))));
+        tracks.push(track);
+    }
+
+    release.discs.push({
+        tracks: tracks,
+        format: release.format,
+    });
+
+    // define releasing label
+    release.labels.push(translateLabel(release_heading.childNodes[2].innerText.replace(/\s/g,' '), catno));
+
+    LOGGER.info('Parsed release: ', release);
+    return release;
+}
+
+// Insert button into page under label information
+function insertLink(release, release_url) {
+    let edit_note = MBImport.makeEditNote(release_url, 'HardTunes.com');
+    let parameters = MBImport.buildFormParameters(release, edit_note);
+
+    document.getElementsByClassName('social-media')[0].innerHTML += MBImport.buildFormHTML(parameters) + MBImport.buildSearchButton(release);
+}
+
+// Fixed mappings for some popular labels
```

Comment:
I really doubt that having an hardcoded mapping is a good idea (both for labels and artists): it's error prone and difficult to maintain.

Reply:
This online shop has specialized on a few genres and name clashes within closely related genres are quite unlikely.
But I have no plans for extending that list any further, it’s only meant for more popular artists, which might be on compilations more.